torch/_dynamo/variables/torch.py

```diff
@@ -1177,6 +1183,31 @@ def speculate_branch(branch):
             )
             r = body_r.as_proxy().node.meta["example_value"]
             example_value = r
+        elif self.value.__name__ == "trampoline_autograd_fn":
+            fn = TorchVariable(
+                self.value
+            )
+
+            checkpoint = tx.copy_graphstate()
+            graph_checkpoint = tx.output.graph
+            (
+                body_r,
+                body_graph,
+                body_lifted_freevars,
+            ) = speculate_subgraph(
+                fn,
+                [
+                    *args,
+                ],
+                graph_checkpoint,
+                checkpoint,
+            )
+            p_args = (
+                *(arg.as_proxy() for arg in args),
+                *(arg for arg in body_lifted_freevars),
+            )
```

Comment:
It looks like we will lift freevars to being inputs of `trampoline_autograd_fn`. Later on, `trampoline_autograd_fn(*args, **kwargs)` will call `autograd_function.apply(*args, **kwargs)`

We haven't actually re-written the autograd.Function to accept more inputs, so this seems wrong. We should probably just graph break if we see free variables (or figure out how to re-install the free variables in trampoline_autograd_fn)

Reply:
Good point.